src/js/midi/midi.js

```diff
@@ -302,23 +293,22 @@ jQuery(() => {
     .on('note off', (note, velocity, channel) => {
       synth.noteOff(note, true)
     })
+    .on('update', () => {
+      if (state.get('midi modal visible')) {
+        state.set('midi modal visible', false)
+        state.set('midi modal visible', true)
```

Comment:
Is this evoking some state mutation magic? A clarifying comment could help. At the outset it looks like it does nothing.

Reply:
Yes, this is a bit hacky. The midi modal listens to the changes of this variable and renders the template when the state goes from false to true. The state manager only sends out a change event when the value changes, so here I'm forcing a change with the value true to make the modal close and re-render

Reply:
Normally, if there would be react in the codebase or any proper templating solution, then it would update the DOM based on the props given to the modal, but for now I have to do it manually